Develop

BiroAron opened the pull request:
Second part of the task.

BiroAron committed:
A feladat elso resze

BiroAron committed:
Removed HelloWorld.vue file and separated ToDoAdd component from ToDoList component

komaromiGyorgy commented on src/components/ToDoAdd.vue:
```diff
@@ -0,0 +1,36 @@
+  </div>
+</template>
+
+<script lang="ts">
```
Can you use script setup in all components, please?

BiroAron committed:
Second part of the task

BiroAron committed:
Second part corrected

Added new components. Removed helloword.wue and some unnecessary css.

roli-wg commented:
The pull request title HAS TO BE the title from the trello ticket which you finish with this PR (including the project code ITL and the ticket number in this case 1) also the commit messages should be the subtask from the task
From now we will not check the code you pushed until the PR doesn't match this circumstances!

BiroAron committed:
ITL-7 Create the header component

BiroAron committed:
ITL-8 Add the title

BiroAron committed:
ITL-9 Add an add button, include a icon library or include the icon from the design

BiroAron committed:
ITL-10 Create the event that triggers the todo creation functionality

BiroAron committed:
Replaced plain CSS with Tailwind

BiroAron committed:
Made all the changes regarding to the comments

BiroAron committed:
Made the changes regarding to class names and reactive variables

BiroAron committed:
Removed all unnecessary code and comments

BiroAron committed:
Removed unnecessary id-s and redundant script

BiroAron committed:
Merge branch 'feature/ITL-1-implement-page-header' into develop

BiroAron committed:
ITL 11 Create the Placeholder img when the user is in the home page and no todos added to todo list

BiroAron committed:
ITL 12 Mobile friendly looking

BiroAron committed:
ITL 13  make the todo list look

BiroAron committed:
Made changes regarding the comments

-removed all of the conditional operators from the template tags
-removed unnecessary complicated code
-renamed non-descriptive variable names
-created a colorMap

BiroAron committed:
updatePriorityClass is now a variable instead of a function

BiroAron committed:
Merge branch 'feature/ITL-2-implement-todo-list' into develop

BiroAron committed:
ITL14 Subtask : Create an empty to-do item

BiroAron committed:
ITL 15Subtask 2: Display the new to-do item at the top of the list

BiroAron committed:
ITL 16Subtask 3: Set proper placeholder text for the new to-do item

BiroAron committed:
Created the requested file system

BiroAron committed:
ITL 17 Subtask 4 :After tapping the task, Card will convert it into edit mode

BiroAron committed:
ITL 19  Subtask 6 : Make the design match with figma

BiroAron committed:
ITL 20 Subtask 7: Pop-up confirmation

BiroAron committed:
Cleaned up the code

BiroAron committed:
Made changes regarding the comments

- removed modification of props in child components

- component API orders corrected

BiroAron committed:
Removed all reactive usage of props in child elements

BiroAron committed:
Divided the ToDoForm.vue into five subcomponents

BiroAron committed:
Merge branch 'feature/ITL-3-implement-edit-and-add-state-for-todo-card' into develop

BiroAron committed:
ITL 21 move the completed todo item to the bottom of the todo list

when the user completes a todo ,animate a checkmark  and move the completed todo item to the bottom of the todo list.

BiroAron committed:
Made changes regarding the comments and fixed reactivity issues

BiroAron committed:
ITL 4 - Made changes regarding the comments

-removed components named ToDoDeleteButton and TodoSaveButton components, combined them into one named Button.vue

BiroAron committed:
ITL 4 - Made changes regarding the comments

- replaced Button.vue with BaseButton.vue
- replaced :value with v-model in TodoDescription and TodoTitle components
- used v-for instead of creating multiple li elements in TodoPriority component

BiroAron committed:
ITL 4 - Replaced all  basic button elements with BaseButton.vue

BiroAron committed:
Merge branch 'feature/ITL-4-implement-base-state-of-todo-card' into develop

BiroAron committed:
ITL 22 - Make an input field and design it

BiroAron committed:
ITL-23 - Make the input field to work, search the todo list

BiroAron committed:
Merge branch 'feature/ITL-5-implement-search-functionality' into develop

BiroAron committed:
ITL 24 - Implement make  a component and six button first

BiroAron committed:
ITL 25 - Implement the ability to sort by title

BiroAron committed:
ITl 26 - Implement the ability to sort by due date

BiroAron committed:
ITL 27 - Implement the ability to sort by priority

BiroAron committed:
ITL 28, ITL 29 - Implement the sort in ascending/descending order

BiroAron committed:
ITL 6 - Made some additional changes

- saving todos list to localstorage
- hiding searchbar and filtering buttons when they aren't necessary

BiroAron committed:
ITL 6 - Made some changes regarding the comments

- corrected typos in parameter names
- deleted unnecessary else statements
- restructured Filter.vue according to the style guide
- used v-if v-else where possible

BiroAron committed:
ITL 6 - Made some additional changes

 - corrected priority buttons
- modified Todo type: todo.text is now todo.description

BiroAron committed:
ITL 6 - Added hover effect to todo elements, fixed delete blur bug

BiroAron committed:
ITL 6 -  Corrected css

BiroAron committed:
ITL 6 - Corrected some more css and fixed a v-if

BiroAron committed:
Merge branch 'feature/ITL-6-add-sorting-functionality' into develop

BiroAron committed:
Finalizing - made changes regarding the feedback at the demo

- added a smaller text size to phone view
- made two separate sort direction setting buttons

BiroAron committed:
Finalizing - adjusted class orders

BiroAron committed:
Merge branch 'feature/Finalizing-code-and-resolvig-issues' into develop

BiroAron committed:
ITL 30 - Created a new file structure

- created a new file structure
- initialized the new project

BiroAron committed:
ITL 8 - Fixed the file structure

BiroAron committed:
Merge branch 'feature/ITL-8-add-the-basic-NodeJS-project-into-a-folder-in-the-current-TodoApp' into develop

BiroAron committed:
ITL 31 - Add todo model

BiroAron committed:
ITL 32 - Add users model

BiroAron committed:
ITL 7 - Made changes regarding the comments

-changed the type of date in the todos model
-exported without creating variables

BiroAron committed:
Merge branch 'feature/TL-7-create-the-DB-structure' into develop

BiroAron committed:
ITL 33 - Add a Login page

BiroAron committed:
ILT 34 - Add a Sign Up page

BiroAron committed:
ITL 35 - Create routes

BiroAron committed:
ITL 9 - Made some changes on the frontend

- made changes to the frontend so it uses the operations on the backend

BiroAron committed:
ITL 9 - Restructured the backend

BiroAron committed:
ITL 9 - Made a reusable input field and used it

BiroAron committed:
ITL 9 - Made changes regarding the comments

- removed commented lines an unnecesary consol.log-s
- used v-model instead of passing props into Baseinput.vue
- replaced fetchAllTodos function where it was possible

BiroAron committed:
ITL 9 - Made some additional changes

- removed unnecessary functions from signup and login pages

BiroAron committed:
ITL 9 - Made the last change

- used TodoPriorities enum to give a default value to the priority of a todo

BiroAron committed:
ITL 9 - Made changes regarding the comments

- added a redirect to the router
- added custom color to tailwind config
- in addition also fixed a bug regarding the display of the date

BiroAron committed:
ITL 9 - Created a logout button

BiroAron committed:
ITL 9 - Made all the changes requested

BiroAron committed:
ITL 9 - Added full routing on server side

BiroAron committed:
ITL 9 - Made some minor fixes

BiroAron committed:
ITL 9 - Made all the changes requested on the pr

BiroAron committed:
Merge branch 'feature/TL-9-implement-multiple-page-todo-app' into develop

BiroAron committed:
ITL 33 - Create CRUD operations

- also moved search and filtering functionality
- implemented archive by making soft deletes

BiroAron committed:
ITL 34 - Implemented authentication

BiroAron committed:
ITL 10 - Removed all the console.log()-s

BiroAron committed:
ITL 10 - Switched switch

BiroAron committed:
ITL 10 - Made requested chnages

BiroAron committed:
ITL 10 - Sorting and filtering is now done with a pipeline

BiroAron committed:
ITL 10 - Added validations before calling services

BiroAron committed:
ITL 10 - Corrected mistakes and got rid of unnecessary code

BiroAron committed:
ITL 10 - Removed console log

BiroAron committed:
ITL 10 - Fixed small issues

BiroAron committed:
Merge branch 'feature/ITL-10-create-backend-files' into develop

BiroAron committed:
ITL 35 - Send less request when creating a new todo and new user

BiroAron committed:
ITL 36 - Remove filter_by frield the query if it doesn't have a value

BiroAron committed:
ITL 37 - Don't have to specify everywhere the contents of a header

BiroAron committed:
ITL 38 - Implement guard on the routes

BiroAron committed:
ITL 39 - Correct the design of the login/sign-up

BiroAron committed:
ITL 11 - Removed unnecessary code and modified FetchTodos function

BiroAron committed:
ITL 11 - Made changes regarding the comments

BiroAron committed:
ITL 11 - Removed unused and unnecessary variables

BiroAron committed:
Merge branch 'feature/ITL-11-make-changes-regarding-the-notes-on-the-demo' into develop

aron-biro committed:
DBchange

aron-biro committed:
Merge branch 'feature/DBchange' into develop
